### PR TITLE
feat(removeLogs): added function to remove within visual selection

### PR DESCRIPTION
**Pull request by VBenny42**
## Checklist
- [X] Used only camelCase variable names.
- [X] If functionality is added or modified, also made respective changes to the
  `README.md` (the `.txt` file is auto-generated and does not need to be modified).

I added a function to remove logs only for selected text. My use case is removing logs for specific functions only. I wanted to make it work as an operator function but that was more complex than I thought.

### CODE REVIEW

**Comment by chrisgrieser** [comment]
Good idea. 


I think it will reduce code repetition as well as make the feature more intuitive to use if you made it into one command though?

Just have `.removeLogs` change its behavior when called in visual mode.

File: lua/chainsaw/core/log-commands.lua [hunk 1]
```diff
@@ -104,5 +104,38 @@ function M.removeLogs()
...
+
+	-- Get the start and end of current visual selection
+	local selectedStart = vim.fn.getpos("v")
+	local selectedEnd = vim.fn.getpos(".")
+	if selectedStart[2] > selectedEnd[2] then
+		selectedStart, selectedEnd = selectedEnd, selectedStart
+	end
+
+	-- Remove lines only for visual selection
+	local bufLines = vim.api.nvim_buf_get_lines(0, selectedStart[2] - 1, selectedEnd[2], false)
```
Comment (chrisgrieser) [inline lua/chainsaw/core/log-commands.lua:119]: Shouldn't this var be named `selectedLine`? Those aren't all lines of the buffet after all

File: lua/chainsaw/core/log-commands.lua [hunk 2]
```diff
@@ -104,5 +104,38 @@ function M.removeLogs()
...
+	end
+
+	-- notify on number of lines removed
+	local linesRemoved = numOfLinesBefore - vim.api.nvim_buf_line_count(0)
+	local msg = ("Removed %d lines."):format(linesRemoved)
+	if linesRemoved == 1 then msg = msg:sub(1, -3) .. "." end -- 1 = singular
+	require("chainsaw.utils").info(msg)
+
+	-- Go back to normal mode
+	vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes("<Esc>", true, false, true), "v", true)
```
Comment (chrisgrieser) [inline lua/chainsaw/core/log-commands.lua:134]: It's simpler to go back to normal mode by calling `vim.cmd.normal {vim.fn.mode(), bang = true}`

File: lua/chainsaw/core/log-commands.lua [hunk 3]
```diff
@@ -104,5 +104,38 @@ function M.removeLogs()
...
+	local selectedEnd = vim.fn.getpos(".")
+	if selectedStart[2] > selectedEnd[2] then
+		selectedStart, selectedEnd = selectedEnd, selectedStart
+	end
+
+	-- Remove lines only for visual selection
+	local bufLines = vim.api.nvim_buf_get_lines(0, selectedStart[2] - 1, selectedEnd[2], false)
+	for i = #bufLines, 1, -1 do
+		if bufLines[i]:find(marker, nil, true) then
+			local actualLine = selectedStart[2] + i - 1
```
Comment (chrisgrieser) [inline lua/chainsaw/core/log-commands.lua:122]: This var might better be named `lnum`

File: lua/chainsaw/core/log-commands.lua [hunk 4]
```diff
@@ -104,5 +104,38 @@ function M.removeLogs()
 	vim.b.timelogStart = nil
 end
 
+function M.removeLogsVisual()
+	local marker = require("chainsaw.config.config").config.marker
+	local numOfLinesBefore = vim.api.nvim_buf_line_count(0)
+
+	-- Get the start and end of current visual selection
+	local selectedStart = vim.fn.getpos("v")
+	local selectedEnd = vim.fn.getpos(".")
```
Comment (chrisgrieser) [inline lua/chainsaw/core/log-commands.lua:113]: Just a bit, but directly storing `[2]` here would make the code more readable; `[1]` is never accessed after all

**Review by chrisgrieser — CHANGES_REQUESTED** [review]
No deal breakers, just a few minor changes to improve readability

File: lua/chainsaw/core/log-commands.lua [hunk 5]
```diff
@@ -104,5 +104,38 @@ function M.removeLogs()
...
+	if selectedStart[2] > selectedEnd[2] then
+		selectedStart, selectedEnd = selectedEnd, selectedStart
+	end
+
+	-- Remove lines only for visual selection
+	local bufLines = vim.api.nvim_buf_get_lines(0, selectedStart[2] - 1, selectedEnd[2], false)
+	for i = #bufLines, 1, -1 do
```
Comment (chrisgrieser) [inline lua/chainsaw/core/log-commands.lua:120]: How about running this loop from selectedEnd to selectedEnd? That way the iterator is automatically already the line number.

**Review by chrisgrieser — CHANGES_REQUESTED** [review]

**Comment by VBenny42** [comment]
Okiedoke, I've modified removeLogs as needed. Let me know if there's any other changes.

**Comment by chrisgrieser** [comment]
Thx, LGTM

**Pull request merged**